lib/cognito-idp-stack.ts

```diff
@@ -0,0 +1,380 @@
+
+        // Create the REST API with an L3 construct included in this example repo.
+        // (See cognito-rest-api.ts)
+        const api = new CognitoRestApi(this, this.stackName, {
+            domainName: props.apiDomainName,
+            certificateArn: props.apiCertificateArn,
+            lambdaFunctionDirectory: './dist/lambda',
+            userPool,
+            cognitoRedirectUri: `https://${props.webDomainName}`,
+            cognitoDomainPrefix,
+            cognitoAppClientId: userPoolClient.userPoolClientId,
+            cognitoRegion: region,
+            additionalEnvVars: {
+                "USER_TABLE": userTable.tableName
+            },
+            resourceHandlers: handlers, 
+            hostedZoneId: props.apiHostedZoneId
+        });
+
+        // Static web site created by an L3 construct included in this example repo
+        // (See static-site.ts)
+        const site = new StaticSite(this, 'StaticSite', {
+            domainName: props.webDomainName,
+            certificateArn: props.webCertificateArn,
+            contentPath: './dist/web', 
+            hostedZoneId: props.hostedZoneId
+        });
+
+        // Create a custom resource that writes out the config file for the web site.
+        // (The web site needs deploy-time values, so this fixes some of the chicken 
+        // and egg problems with the .env file)
+        const onEvent = new lambda.Function(this, 'CreateConfigHandler', {
+            runtime: lambda.Runtime.NODEJS_12_X,
+            code: lambda.Code.fromAsset('./dist/lambda'),
+            handler: `create-config.handler`,
+            memorySize: 1536,
+            timeout: cdk.Duration.minutes(5),
+            description: `${this.stackName} Static Site Config`,
+            environment: {
+                'S3_BUCKET_NAME': site.getBucketName(),
+                'API_DOMAIN': props.apiDomainName,
+                'COGNITO_DOMAIN_PREFIX': cognitoDomainPrefix,
+                'COGNITO_REGION': region,
+                'COGNITO_APP_CLIENT_ID': userPoolClient.userPoolClientId,
+                'COGNITO_REDIRECT_URI': props.cognitoRedirectUri,
+                'FACEBOOK_APP_ID': props.facebookAppId,
+                'FACEBOOK_VERSION': props.facebookApiVersion
+            }
+        });
+
+        site.grantAccessTo(onEvent);
+
+        // Create a provider
+        const provider = new cr.Provider(this, 'ConfigFileProvider', {
+            onEventHandler: onEvent
+        });
+
+        // Create the custom resource
+        const customResource = new cdk.CustomResource(this, 'ConfigFileResource', {
+            serviceToken: provider.serviceToken, 
+            properties: { 
+                'FORCE_UPDATE': new Date().toISOString()
+            }
+        });
+
+        // FORCE_UPDATE forces the custom resource to update the config file on each deploy
+
+        // TODO - Can we set the logical id of the custom resource every time the deployment changes?
```

Comment:
What's this todo?

Reply:
That's to try and figure out a better way to force the config file to be created only when the bucket deployment happens, since it wipes out the config. As it is, I force the custom resource to run every single time a deployment happens, even if the web site didn't change.